exporters/stdout TestStdoutTimestamp test failure

This test seems to consistently be failing for Windows testing using Go 1.14 and 1.15:

https://github.com/open-telemetry/opentelemetry-go/runs/1962753557?check_suite_focus=true

```
go test -timeout 30s -short ./exporters/stdout/...
--- FAIL: TestStdoutTimestamp (0.04s)
    metric_test.go:122: 
        	Error Trace:	metric_test.go:122
        	Error:      	Should be true
        	Test:       	TestStdoutTimestamp
FAIL
FAIL	go.opentelemetry.io/otel/exporters/stdout	0.775s
FAIL
mingw32-make: *** [Makefile:100: test] Error 123
```

Looking at the test ...

https://github.com/open-telemetry/opentelemetry-go/blob/b94cd4b24917d0099eb0ddee6f9d6be0bddee469/exporters/stdout/metric_test.go#L78-L124

## Proposals

- It is testing a "before" time is before a measured timestamp without ensuring there is a delay of time. We should add a small artificial duration to ensure this test succeeds.
- We can remove the check of the before/after timestamp. I'm not sure these are actually adding value. I think we need to verify that a timestamp is returned, but I'm not sure it is the exporters responsibility to ensure the monotonicity of the timestamps (rather that is the metrics pipeline itself).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -31,6 +31,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Removed attempt to resample spans upon changing the span name with `span.SetName()`. (#1545)
 
+### Fixed
+
+- The sequential timing check of timestamps in the stdout exporter are now setup explicitly to be sequential (#1571). (#1572)
+
 ## [0.17.0] - 2020-02-12
 
 ### Changed

--- a/exporters/stdout/metric_test.go
+++ b/exporters/stdout/metric_test.go
@@ -85,6 +85,8 @@ func TestStdoutTimestamp(t *testing.T) {
 	}
 
 	before := time.Now()
+	// Ensure the timestamp is after before.
+	time.Sleep(time.Nanosecond)
 
 	checkpointSet := metrictest.NewCheckpointSet(testResource)
 
@@ -102,6 +104,8 @@ func TestStdoutTimestamp(t *testing.T) {
 		t.Fatal("Unexpected export error: ", err)
 	}
 
+	// Ensure the timestamp is before after.
+	time.Sleep(time.Nanosecond)
 	after := time.Now()
 
 	var printed []interface{}

